web-bundle/src/main/java/com/graphhopper/resources/MapboxResource.java

```diff
@@ -0,0 +1,185 @@
+            if (!enableInstructions)
+                throwIllegalArgumentException("Currently, you need to enable steps");
+            if (!roundaboutExits)
+                throwIllegalArgumentException("Roundabout exits have to be enabled right now");
+            if (!voiceUnits.equals("metric"))
+                throwIllegalArgumentException("Voice units only support metric right now");
+            if (!voiceInstructions)
+                throwIllegalArgumentException("You need to enable voice instructions right now");
+            if (!bannerInstructions)
+                throwIllegalArgumentException("You need to enable banner instructions right now");
+
+            double minPathPrecision = 1;
+            if (overview.equals("full"))
+                minPathPrecision = 0;
+
+            String vehicleStr = convertProfileToGraphHopperVehicleString(profile);
+            List<GHPoint> requestPoints = getPointsFromRequest(pathSegment);
+
+            StopWatch sw = new StopWatch().start();
+
+            // TODO: initialization with heading/bearings
+            // TODO: how should we use the "continue_straight" parameter? This is analog to pass_through, would require disabling CH
+
+            GHRequest request = new GHRequest(requestPoints);
+            request.setVehicle(vehicleStr).
+                    setWeighting(weighting).
+                    setLocale(localeStr).
+                    getHints().
+                    put(CALC_POINTS, true).
+                    put(INSTRUCTIONS, enableInstructions).
+                    put(WAY_POINT_MAX_DISTANCE, minPathPrecision);
+
+            GHResponse ghResponse = graphHopper.route(request);
+
+            float took = sw.stop().getSeconds();
+            String infoStr = httpReq.getRemoteAddr() + " " + httpReq.getLocale() + " " + httpReq.getHeader("User-Agent");
+            String logStr = httpReq.getQueryString() + " " + infoStr + " " + requestPoints + ", took:"
+                    + took + ", " + weighting + ", " + vehicleStr;
+
+            if (ghResponse.hasErrors()) {
+                logger.error(logStr + ", errors:" + ghResponse.getErrors());
+                // Mapbox specifies 422 return type for input errors
+                return Response.status(422).entity(MapboxResponseConverter.convertFromGHResponseError(ghResponse)).
+                        header("X-GH-Took", "" + Math.round(took * 1000)).
+                        build();
+            } else {
+                return Response.ok(MapboxResponseConverter.convertFromGHResponse(ghResponse, request.getLocale())).
+                        header("X-GH-Took", "" + Math.round(took * 1000)).
+                        build();
+            }
+
+        } catch (Exception e) {
+            logger.error("Exception while processing: ", e);
+            throw new MultiException(e);
+        }
+    }
+
+    private void throwIllegalArgumentException(String exceptionMessage) {
+        logger.error(exceptionMessage);
```

Comment:
Why do we need this method? An exception should be logged from dropwizard automatically

Reply:
Thanks, I will try this again, but IIRC in my tests so far, errors have not been logged by dropwizard, that's why I added this :).

Reply:
Sorry, IllegalArgEx are supposed to avoid extra logging. But still it should print the request and info about a 'bad request'. If not, then we need to fix it globally for all IllegalArgEx

Reply:
The exception should not be logged separately (please just throw an IllegalArgumentException or if not a bad request but indicating an error throw RuntimeException or IllegalState or whatever). 

If there is a bad request and it is not logged at all, we have to fix it for all resources, not sure here.

Reply:
Ok, will simply change this :)